moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -954,115 +881,86 @@ bool TimeOptimalTrajectoryGeneration::computeTimeStamps(
         return false;
       }
-      max_velocity[idx] =
+      max_velocity[j] =
           std::min(std::fabs(bounds.max_velocity_), std::fabs(bounds.min_velocity_)) * velocity_scaling_factor;
-      set_velocity_limit = true;
     }
-
-    if (!set_velocity_limit)
+    else  // default limit
     {
-      ROS_ERROR_STREAM_NAMED(LOGNAME, "No velocity limit was defined for joint "
-                                          << vars[idx].c_str()
-                                          << "! You have to define velocity limits in the URDF or "
-                                             "joint_limits.yaml");
-      return false;
+      max_velocity[j] = DEFAULT_VELOCITY_LIMIT;
+      ROS_WARN_NAMED(LOGNAME, "No velocity limits defined for '%s'! Define them in URDF or joint_limits.yaml",
+                     name.c_str());
     }
 
     // ACCELERATION LIMIT
-    // Check if a custom limit was specified as an argument
-    bool set_acceleration_limit = false;
-    it = acceleration_limits.find(vars[idx]);
+    it = acceleration_limits.find(name);  // check for a custom limit
     if (it != acceleration_limits.end())
     {
-      max_acceleration[idx] = it->second * acceleration_scaling_factor;
-      set_acceleration_limit = true;
+      max_acceleration[j] = it->second * acceleration_scaling_factor;
     }
 
-    if (bounds.acceleration_bounded_ && !set_acceleration_limit)
+    else if (bounds.acceleration_bounded_)  // resort to the limit from the robot model
     {
-      // Set the default acceleration limit, from robot model
-      if (bounds.max_acceleration_ < 0.0)
+      if (bounds.max_acceleration_ <= std::numeric_limits<double>::epsilon())
       {
         ROS_ERROR_NAMED(LOGNAME, "Invalid max_acceleration %f specified for '%s', must be greater than 0.0",
-                        bounds.max_acceleration_, vars[idx].c_str());
+                        bounds.max_acceleration_, name.c_str());
         return false;
       }
-      max_acceleration[idx] = std::min(std::fabs(bounds.max_acceleration_), std::fabs(bounds.min_acceleration_)) *
-                              acceleration_scaling_factor;
-      set_acceleration_limit = true;
+      max_acceleration[j] = std::min(std::fabs(bounds.max_acceleration_), std::fabs(bounds.min_acceleration_)) *
+                            acceleration_scaling_factor;
     }
-    if (!set_acceleration_limit)
+    else  // default limit
     {
-      ROS_ERROR_STREAM_NAMED(LOGNAME, "No acceleration limit was defined for joint "
-                                          << vars[idx].c_str()
-                                          << "! You have to define acceleration limits in the URDF or "
-                                             "joint_limits.yaml");
-      return false;
+      max_acceleration[j] = DEFAULT_ACCELERATION_LIMIT;
+      // ROS_WARN_NAMED(LOGNAME, "No acceleration limits defined for '%s'! Define them in joint_limits.yaml", name.c_str());
```

Comment:
Unless you had a special reason for this...

```suggestion
      ROS_WARN_NAMED(LOGNAME, "No acceleration limits defined for '%s'! Define them in joint_limits.yaml", name.c_str());
```

Reply:
I deliberatively commented that out, because we didn't warn about missing acceleration limits in the past.
However, I'm fine to include it as well.

Reply:
I think we should throttle those warnings to `ONCE`. We don't need to repeat the same info over and over for all joints.

Reply:
fine with me